### PR TITLE
 Convert to AWS-supported Lambda calling convention; fix/add tests

### PR DESCRIPTION
**Pull request by otterley**
ebs-backup now conforms to the [official AWS Lambda model for Go](https://docs.aws.amazon.com/lambda/latest/dg/go-programming-model-handler-types.html). 

I've also fixed broken tests, and added a full end-to-end test suite that exercises an actual snapshot from an EC2 instance and validates the tags.

### CODE REVIEW

File: functions/ebs-backup/main.go [hunk 1]
```diff
@@ -28,32 +25,55 @@ var env = []string{
...
+			Name:       e.Name,
+			SnapshotID: res.CreatedSnapshot,
+			VolumeID:   res.VolumeID,
 		}
+		if res.Err == nil {
+			log.WithFields(fields).Info("snapshot")
+		} else {
+			errOccurred = true
+			fields["error"] = res.Err.Error()
+			log.WithFields(fields).Error("snapshot")
```
Comment (yields) [inline functions/ebs-backup/main.go]: WithError(err)?

File: functions/ebs-backup/main.go [hunk 2]
```diff
@@ -28,32 +25,55 @@ var env = []string{
...
+			result.Error = res.Err.Error()
+		}
+		r = append(r, result)
+	}
 
-		return results, nil
-	})
+	if errOccurred {
+		return r, errors.New("An error occurred")
```
Comment (yields) [inline functions/ebs-backup/main.go]: Would returning the actual error, or a list of errors be a better choice? or does the caller lose the error msg anyway?

Comment (otterley) [reply functions/ebs-backup/main.go]: The actual errors are logged above (see `fields["error"]`), so this is just some generic text.

File: functions/ebs-backup/main.go (hunk 1 above)
Comment (otterley) [reply functions/ebs-backup/main.go]: I want to log the fields, but at an Error severity level.  Make sense?

**Pull request merged**